charm/toolbox/ecgroup.py

```diff
@@ -58,32 +58,40 @@ def serialize(self, object):
     def deserialize(self, bytes_object):
         """deserializes into a pairing object"""        
         return deserialize(self.ec_group, bytes_object)
-    
+
     def hash(self, args, target_type=ZR):
-        """hashes objects into ZR or G"""        
-        if isinstance(args, tuple):
-            s = bytes()
-            DEL = '|'
-            for i in args:
-                if type(i) == ec_element:
-                    s += serialize(i)
-                elif type(i) == str:
-                    s += bytes(str(i) + DEL, 'utf8')
-                elif type(i) == bytes:
-                    s += i + bytes(DEL, 'utf8') 
-                else:
-                    print("unexpected type: ", type(i))
-                # consider other types    
-            #print("s => %s" % s)
-            assert len(s) != 0, "hash input is empty."
```

Comment:
This assertion doesn’t exists anymore in the new code. Now `group.hash('') == group.hash(tuple())`.